### ISSUE TITLE
[java] UnusedPrivateField: Do not flag fields annotated with @Id or @EmbeddedId

### ISSUE THREAD

**Issue by numeralnathan** [pmd/pmd#3825]
**Affects PMD Version:** 6.43.0

**Rule:** [UnusedPrivateField](https://pmd.github.io/pmd-6.43.0/pmd_rules_java_bestpractices.html#unusedprivatefield)

**Description:**

Fields annotated with `@javax.persistence.Id` or `@javax.persistence.EmbeddedId` are likely to not have a getter or setter.  Please fix `UnusedPrivateField` to ignore fields with these annotatations.

**Code Sample demonstrating the issue:**

```
@Entity
@Table(name = "my_table")
public class MyTable implements Serializable
{
   private static final long serialVersionUID = 1L;

   @Id
   @Column(name = "id", nullable = false)
   private long m_id;   // PMD Violation: UnusedPrivateField

   // other fields

   /*
    * For JPA
    */
   public MyTable()
   {
      // nothing to do
   }

   // getters and setters for other fields except m_id
}
```

**Expected outcome:**

PMD reports a violation at line `private long m_id;   // PMD Violation: UnusedPrivateField`, but that's wrong. That's a false positive.

**Running PMD through:** Eclipse PMD Plugin

**Comment by pzygielo** [comment]
I think you can customize the rule with `ignoredAnnotations`, as this is user-configurable list.

**Comment by oowekyala** [comment]
@pzygielo Thanks for pointing this out. It's true that this is configurable, but I wonder what for. Are there any cases where an annotation should *not* cause the field be ignored by the rule? I think `@Deprecated` and `@SuppressWarnings` should maybe get a pass, but any other annotation should suppress the violation. This is related to #3791, where I was suggesting doing the same to UnnecessaryConstructor.

**Comment by pzygielo** [comment]
> I think `@Deprecated` and `@SuppressWarnings` should maybe get a pass, but any other annotation should suppress the violation.

This was raised specifically for `@Id` and `@EmbeddedId` here and **separately** for `@Version` in #3824.

This
> Please fix

sounds to me like a report for a bug. Which I do not see as being there.

Perhaps both issues could be merged to differently phrased request to modify the rule as [described above](https://github.com/pmd/pmd/issues/3825#issuecomment-1059978582).
Before anything is modified in PMD, there is a way to have ignored annotations extended with current release of the tool.

However - should *this one* be picked for _fixing_ - I suggest to include `@[Embedded]Id` from `jakarta` namespace as well, and the same for `@Version` in #3824.
